app/models/solidus_subscriptions/installment.rb

```diff
@@ -5,6 +5,7 @@
 # successful or otherwise at fulfilling this installment
 module SolidusSubscriptions
   class Installment < ApplicationRecord
+    before_create :void_failed_past_installments, if: ->{ SolidusSubscriptions.configuration.ignore_past_failed_installments_on_new_cycle? }
```

Comment:
Do you think there could be any issues with running this in a callback? Like someone attempting to backfill old data and inadvertently clearing installments as a result?

I feel like this should be done explicitly, perhaps in the `Processor`, when a new installment is created.

Reply:
I've moved the logic in the processor `new_installments` method.